sdk/keyvault/azure-security-keyvault/azure/security/keyvault/keys/_client.py

```diff
@@ -3,266 +3,715 @@
-        request = HttpRequest("POST", url, headers)
-        request.set_json_body(body)
-        request.format_parameters({"api-version": self.API_VERSION})
+        return Key._from_key_bundle(bundle)
 
-        response = self._pipeline.run(request, **kwargs).http_response
-        if response.status_code != 200:
-            raise HttpResponseError("Request failed with code {}: '{}'".format(response.status_code, response.text()))
+    def create_key(
+        self, name, key_type, key_ops=None, enabled=None, expires=None, not_before=None, tags=None, **kwargs
+    ):
+        # type: (str, str, Optional[List[str]], Optional[bool], Optional[datetime], Optional[datetime], Optional[Dict[str, str]], Mapping[str, Any]) -> Key
+        """Creates a new key, stores it, then returns key attributes to the client.
 
-        key = self._deserialize("Key", response)
+        The create key operation can be used to create any key type in Azure
+        Key Vault. If the named key already exists, Azure Key Vault creates a
+        new version of the key. It requires the keys/create permission.
 
-        return key
+        :param name: The name for the new key. The system will generate
+         the version name for the new key.
+        :type name: str
+        :param key_type: The type of key to create. For valid values, see
+         JsonWebKeyType. Possible values include: 'EC', 'EC-HSM', 'RSA',
+         'RSA-HSM', 'oct'
+        :type key_type: str or ~azure.security.keyvault._generated.v7_0.models.JsonWebKeyType
+        :param key_ops: Supported key operations.
+        :type key_ops: list[str or
+         ~azure.security.keyvault._generated.v7_0.models.JsonWebKeyOperation]
+        :param enabled: Determines whether the object is enabled.
+        :type enabled: bool
+        :param expires: Expiry date of the key  in UTC.
+        :type expires: datetime.datetime
+        :param not_before: Not before date of the key in UTC
+        :type not_before: datetime.datetime
+        :param tags: Application specific metadata in the form of key-value
+         pairs.
+        :type tags: Dict[str, str]
+        :returns: The created key
+        :rtype: ~azure.security.keyvault.keys._models.Key
+
+        Example:
+            .. literalinclude:: ../tests/test_examples_keys.py
+                :start-after: [START create_key]
+                :end-before: [END create_key]
+                :language: python
+                :dedent: 4
+                :caption: Creates a key in the key vault
+        """
+        return self._create_key(
+            name, key_type=key_type, key_ops=key_ops, enabled=enabled, expires=expires, not_before=not_before, tags=tags
+        )
 
-    def delete_key(self, name, **kwargs):
-        # type: (str, Any) -> DeletedKey
-        url = "/".join([self.vault_url, "keys", name])
+    def create_rsa_key(
+        self, name, key_type, size=None, key_ops=None, enabled=None, expires=None, not_before=None, tags=None, **kwargs
```

Comment:
For both Java and .NET we have an extra property called `HSM` which is a boolean and determines the specific type for `key_type`. i.e. https://github.com/Azure/azure-sdk-for-java/pull/3643/files#diff-ea785aff65586d60dd17f1e64631ffacR132

Reply:
It applies for EC keys too